flake.nix

```
@@ -0,0 +1,111 @@
+{
+  description = "A simple terminal UI for git commands";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-25.05";
+    systems.url = "github:nix-systems/default";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-compat.url = "https://flakehub.com/f/edolstra/flake-compat/1.tar.gz";
+    treefmt-nix.url = "github:numtide/treefmt-nix";
+  };
+
+  outputs =
+    inputs@{ flake-parts, systems, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.treefmt-nix.flakeModule
+      ];
+
+      systems = import systems;
+
+      perSystem =
+        {
+          pkgs,
+          ...
+        }:
+        let
+          lazygit = pkgs.buildGoModule rec {
+            pname = "lazygit";
+            version = "0.54.2";
+
+            gitCommit = inputs.self.rev or inputs.self.dirtyRev or "dev";
+
+            src = ./.;
+            vendorHash = null;
+
+            # Disable integration tests that require specific environment
+            doCheck = false;
+
+            nativeBuildInputs = with pkgs; [
+              git
+              makeWrapper
+            ];
+            buildInputs = with pkgs; [ git ];
+
+            ldflags = [
+              "-s"
+              "-w"
+              "-X main.commit=${gitCommit}"
+              "-X main.version=${version}"
+              "-X main.buildSource=nix"
+            ];
+
+            postInstall = ''
+              wrapProgram $out/bin/lazygit \
+                --prefix PATH : ${pkgs.lib.makeBinPath [ pkgs.git ]}
+            '';
+
+            meta = with pkgs.lib; {
+              description = "A simple terminal UI for git commands";
+              homepage = "https://github.com/jesseduffield/lazygit";
+              license = licenses.mit;
+              maintainers = [ "jesseduffield" ];
+              platforms = platforms.unix;
+              mainProgram = "lazygit";
+            };
```

```suggestion
            meta = {
              description = "A simple terminal UI for git commands";
              homepage = "https://github.com/jesseduffield/lazygit";
              license = pkgs.lib.licenses.mit;
              maintainers = [ "jesseduffield" ];
              platforms = pkgs.lib.platforms.unix;
              mainProgram = "lazygit";
            };
```

Read: https://github.com/NixOS/nixpkgs/issues/208242